orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/WaitForTerminatedInstancesTask.groovy

```diff
@@ -0,0 +1,100 @@
+
+import com.netflix.spinnaker.orca.DefaultTaskResult
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.clouddriver.OortService
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import retrofit.RetrofitError
+
+@Slf4j
+@Component
+class WaitForTerminatedInstancesTask extends AbstractCloudProviderAwareTask implements RetryableTask {
+  long backoffPeriod = 1000
+  long timeout = 3600000
+
+  @Autowired
+  OortService oortService
+
+  @Override
+  TaskResult execute(Stage stage) {
+    def instances = stage.context."terminate.instances" as List<Map>
+    if (!instances) {
+      return new DefaultTaskResult(ExecutionStatus.FAILED)
+    }
+
+    // Some platforms kill instances within server groups and recreate them with different IDs - for these, we just
+    // ensure the instance specified no longer exists.
+    // Other platforms  kill instances within server groups and recreate them with the same ID - for these, we can't
+    // just check for the absence of the instance alone. Additionally, we must compare the launch times of the new
+    // instance with that of the killed one.
+    // For instances that are not part of any server group (stand alone instances), we ensure the instance no longer
+    // exists and do not expect it to be recreated.
+    if (allInstancesTerminated(stage) || allInstancesRebooted(stage)) {
+      return new DefaultTaskResult(ExecutionStatus.SUCCEEDED)
+    }
+    return new DefaultTaskResult(ExecutionStatus.RUNNING)
+  }
+
+  /**
+   * @return true iff all instances have disappeared from the search query.
+   */
+  boolean allInstancesTerminated(Stage stage) {
+    def instances = stage.context."terminate.instances" as List<Map>
+    String cloudProvider = getCloudProvider(stage)
+
+    return instances.collect { Map instance ->
+      List<Map> searchResult
+      try {
+        searchResult = oortService.getSearchResults(instance.id as String, "instances", cloudProvider)
+      } catch (RetrofitError e) {
+        log.warn e.message
+      }
+
+      return searchResult?.getAt(0)?.totalMatches == 0
+    }.every { it } // ensure all instances terminated.
+  }
+
+  /**
+   * @return true iff the launch time of each instance ID from clouddriver is greater than the launch times specified in
+   * the stage.
+   */
+  boolean allInstancesRebooted(Stage stage) {
+    def instances = stage.context."terminate.instances" as List<Map>
+    String cloudProvider = getCloudProvider(stage)
+
+    def instanceFromOort
+    return instances.collect { Map instanceFromDeck ->
+      try {
+        instanceFromOort = oortService.getInstance(cloudProvider, stage.context.region as String, instanceFromDeck.id as String)
```

Comment:
`OortService#getInstance` expects `account` as its first argument, not `cloudProvider` (https://github.com/spinnaker/orca/pull/689/files#diff-6b84a45106ad8e7ecccf4462d6d1c462R69)